kubeflow/components/paraphrase.yaml

```diff
@@ -0,0 +1,248 @@
+name: Paraphrase
+description: |
+  Automatic paraphrasing of a dataset
+inputs:
+  - {name: image, description: '', default: ''}
+  - {name: s3_bucket, description: '', default: ''}
+  - {name: owner, description: '', default: ''}
+  - {name: dataset_owner, description: '', default: ''}
+  - {name: task_name, description: '', default: '' }
+  - {name: project, description: '', default: ''}
+  - {name: experiment, description: '', default: ''}
+  - {name: dataset, description: '', default: ''}
+  - {name: s3_input_datadir, description: '', default: ''}
+  - {name: filtering_model, description: '', default: ''}
+  - {name: paraphrasing_model, description: '', default: ''}
+  - {name: skip_generation, description: '', default: ''}
+  - {name: skip_filtering, description: '', default: ''}
+  - {name: keep_original_duplicates, description: '', default: ''}
+  - {name: ignore_context, description: '', default: ''}
+  - {name: additional_args, description: '', default: ''}
+outputs:
+  - {name: s3_output_datadir}
+implementation:
+  container:
+    image: '{{inputs.parameters.image}}'
+    command:
+    - /bin/bash
+    - -ex
+    - -c
+    - |
+      . /opt/genie-toolkit/lib.sh
+
+      parse_args "$0" "image s3_bucket owner dataset_owner task_name project experiment dataset s3_input_datadir s3_output_datadir \
+          filtering_model paraphrasing_model \
+          skip_generation skip_filtering keep_original_duplicates task_name ignore_context" "$@"
+      shift $n
+      #s3_datadir=`cat $s3_datadir`
+      #filtering_model=`cat $filtering_model`
+      
+      /opt/genie-toolkit/sync-repos.sh 
+
+      if [ "$ignore_context" = true ] ; then
+        echo "ignoring context"
+      fi
+
+      set -e
+      set -x
+
+      aws s3 sync "${s3_input_datadir}" input_dataset/
+      aws s3 sync --exclude '*/dataset/*' --exclude '*/cache/*' --exclude 'iteration_*.pth' --exclude '*_optim.pth' "${filtering_model}" filtering_model/
+      aws s3 sync --exclude '*checkpoint*' s3://${s3_bucket}/${paraphrasing_model_full_path} paraphraser/
+
+      mkdir -p output_dataset
+      cp -r input_dataset/* output_dataset
+      
+      s3_output_dir="s3://${s3_bucket}/${dataset_owner}/dataset/${project}/${experiment}/${dataset}-gen/`date +%s`/"
+            
+      mkdir -p `dirname $s3_output_datadir`
```

Comment:
is this intended?

Reply:
This what? This code should be identical to the code in train.yaml which creates an output artifact containing the directory URL to s3.

Reply:
In the line before that you replaced `s3_output_dir` with an s3 path. 
I don't think you intended to mkdir on the parent of s3 path.

In the train.yaml, the actual s3 path is stored in a different variable ( I guess the all CAP is confusing).

Reply:
Nope, one is `s3_output_dir` and the other is `s3_output_datadir`. They are different variables.

Reply:
I missed that. You are right.